corehq/apps/api/tests/test_messaging_event_api.py

```diff
@@ -80,6 +81,61 @@ def test_domain_filter(self):
         ordered_data = json.loads(response.content)['objects']
         self.assertEqual(0, len(ordered_data))
 
+    def test_date_filter_lt(self):
+        dates = self._setup_for_date_filter_test()
+        self._check_date_filtering_response({
+            "date.lt": dates[3].isoformat()
+        }, [d.isoformat() for d in dates[:3]])
+
+    def test_date_filter_lte(self):
+        dates = self._setup_for_date_filter_test()
+        self._check_date_filtering_response({
+            "date.lte": dates[3].isoformat()
+        }, [d.isoformat() for d in dates[:4]])
+
+    def test_date_filter_lte_date(self):
+        """`lte` filter with a date (not datetime) should include data
+        on that day."""
+        dates = self._setup_for_date_filter_test()
+        self._check_date_filtering_response({
+            "date.lte": str(dates[0].date())
+        }, [d.isoformat() for d in dates])
+
+    def test_date_filter_gt(self):
+        dates = self._setup_for_date_filter_test()
+        self._check_date_filtering_response({
+            "date.gt": dates[2].isoformat(),
+        }, [d.isoformat() for d in dates[3:]])
+
+    def test_date_filter_gte(self):
```

Comment:
I don't quite follow this test. Won't all 5 test messages be created on today's date?

Reply:
Yes they will. I added this test for symmetry with `test_date_filter_lte`. When supplying a date without the timestamp the API should include all messages on that date.